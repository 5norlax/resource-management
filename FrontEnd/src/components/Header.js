import React from 'react';

const Header = ({name}) => {
    return (
        <div>
            <hr />
            <h1 className='text-center' >{name}</h1>
            <hr />
        </div>
        )
    }
    
export default Header