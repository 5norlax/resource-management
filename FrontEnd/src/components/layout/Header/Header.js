import React from 'react';

export default function Header({name}) {
    return (
        <div>
            <hr />
            <h1 data-testid="header" className='text-center' >{name}</h1>
            <hr />
        </div>
    )
}