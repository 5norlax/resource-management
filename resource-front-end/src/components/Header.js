import React from 'react';

const Header = () => {
    return (
        <div>
            <img style={{marginTop: '15px', marginLeft: '15px'}} alt="YASH Technologies" src="https://www.yash.com/wp-content/themes/Yash/images/yash-logo.svg" title="YASH Technologies" width="130"/>
            <hr />
        </div>
        )
    }
    
export default Header