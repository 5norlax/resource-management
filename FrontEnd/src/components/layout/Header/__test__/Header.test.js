import React from 'react';
import ReactDOM from 'react-dom'
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Header from '../Header';

const NewHOC = (PassedComponent) => {
  return class extends React.Component {
    render() {
      return (
        <div>
          <PassedComponent {...this.props} />
        </div>
      )
    }
  }
}

afterEach(cleanup);

describe('Test Header', () => {

  it('matches snapshot', () => {
    const tree = renderer.create(<Header />).toJSON()
    expect(tree).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Header />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const Component = NewHOC(Header)
    const { getByTestId } = render(<Header name={'test'} />);
    expect(getByTestId("header").textContent).toBe("test");
  });

});