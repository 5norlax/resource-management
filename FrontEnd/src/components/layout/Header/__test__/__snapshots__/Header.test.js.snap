// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header matches snapshot 1`] = `
<div>
  <hr />
  <h1
    className="text-center"
    data-testid="header"
  />
  <hr />
</div>
`;
